src/main/java/org/broadinstitute/hellbender/tools/dataflow/transforms/markduplicates/MarkDuplicatesUtils.java

```diff
@@ -279,88 +236,14 @@ private static int score(final GATKRead record) {
     }
 
     /**
-     * Struct-like class to store information about the paired reads for mark duplicates.
-     */
-    private static final class PairedEnds {
-        private GATKRead first, second;
-
-        PairedEnds(final GATKRead first) {
-            this.first = first;
-        }
-
-        public static PairedEnds of(final GATKRead first) {
-            return new PairedEnds(first);
-        }
-
-        public PairedEnds and(final GATKRead second) {
-            if (second != null &&
-                    ReadUtils.getStrandedUnclippedStart(first) > ReadUtils.getStrandedUnclippedStart(second)) {
-
-                this.second = this.first;
-                this.first = second;
-            } else {
-                this.second = second;
-            }
-            return this;
-        }
-
-        public String key(final SAMFileHeader header) {
-            return MarkDuplicatesReadsKey.keyForPairedEnds(header, first, second);
-        }
-
-        public GATKRead first() {
-            return first;
-        }
-
-        public GATKRead second() {
-            return second;
-        }
-
-        public int score() {
-            return MarkDuplicates.score(first) + MarkDuplicates.score(second);
-        }
-    }
-
-    /**
-     * Special coder for the PairedEnds class.
+     * Utility class to compare the ordering of GATKReads.
```

Comment:
This comment is actively confusing.  Maybe "GATKRead comparator that compares based on mapping position and followed by SAM flags."